feat(react)!: remove top-level useQuery and useClientDocument exports

IGassmann opened the pull request:
Closes #946

## Summary

- Removes top-level exports of `useQuery`, `useQueryRef`, and `useClientDocument` from `@livestore/react`
- Users must now access these hooks via the store instance: `store.useQuery()`, `store.useClientDocument()`
- Type exports (`UseClientDocumentResult`, `Dispatch`, `SetStateAction`, etc.) remain available at the top level

## Migration

```typescript
// Before
import { useQuery, useClientDocument } from '@livestore/react'
const todos = useQuery(query$)
const [state, setState] = useClientDocument(table)

// After
import { useStore } from '@livestore/react'

const store = useStore(storeOptions) // or via a custom hook wrapping useStore() (e.g. useAppStore())
const todos = store.useQuery(query$)
const [state, setState] = store.useClientDocument(table)
```

## Rationale

This change enforces a cleaner API pattern where queries are always explicitly associated with a store instance, removing ambiguity about which store is being queried.

## Test plan

- [x] TypeScript compilation passes
- [x] Linting passes
- [x] Documentation snippets updated

IGassmann committed:
feat(react)!: remove top-level useQuery and useClientDocument exports

acusti reviewed (commented):
the motivation behind this PR makes sense to me, but i don't get the specific examples. where is `useAppStore` coming from? i only see `declare const useAppStore: ...`

am unimportant aside: "App Store" is a loaded term. but i think it makes sense here, so 🤷

IGassmann committed:
docs: clarify useAppStore usage in snippets

Address PR feedback by importing useAppStore from a shared store
snippet instead of using unexplained type stubs. This makes the
migration examples clearer by showing where useAppStore comes from.

IGassmann committed:
docs: consolidate schema definitions across snippets

- Centralize schema in framework-integrations/react/schema.ts
- Update all snippets to import from the shared schema
- Remove redundant schema.ts files from reactivity-system and columns
- Fix field names to be consistent (newTodoText, filter)
- Add createdAt to todos table and events

IGassmann commented:
> the motivation behind this PR makes sense to me, but i don't get the specific examples. where is `useAppStore` coming from? i only see `declare const useAppStore: ...`
> 
> am unimportant aside: "App Store" is a loaded term. but i think it makes sense here, so 🤷

@acusti Thanks for the feedback! I've updated the snippets to import `useAppStore` from a shared store file (`reference/framework-integrations/react/store.ts`) that shows the actual implementation:

```typescript
export const useAppStore = () =>
  useStore({
    storeId: 'app-root',
    schema,
    adapter,
    batchUpdates,
  })
```

The migration example in the CHANGELOG and PR description now also hints at this pattern with `// or via a custom hook wrapping useStore() (e.g. useAppStore())`.

acusti commented on docs/src/content/_assets/code/reference/framework-integrations/react/schema.ts:
```diff
@@ -7,25 +7,29 @@ export const tables = {
       id: State.SQLite.text({ primaryKey: true }),
       text: State.SQLite.text(),
       completed: State.SQLite.boolean({ default: false }),
+      createdAt: State.SQLite.datetime(),
```
good call on adding `createdAt` to the reference

acusti commented on docs/src/content/_assets/code/reference/reactivity-system/react-component.tsx:
```diff
@@ -1,11 +1,12 @@
 export const MyComponent: FC = () => {
-  const value = useQuery(state$)
+  const store = useAppStore()
+  const value = store.useQuery(state$)
```
this is out of scope (pre-existing and largely unrelated to removing the top-level `useQuery` and `useClientDocument` exports), but this is another code example where the intention isn’t clear to me on what a variable refers to or how i should use it in my own codebase (in this case `state$`)

IGassmann replied:
Thanks! Addressed in 9ab467cba19c592cee4f32fa99faa95f9682f130

acusti commented on CHANGELOG.md:
```diff
@@ -219,6 +219,24 @@
+  // After
+  import { useStore } from '@livestore/react'
+
+  const store = useStore(storeOptions) // or via a custom hook wrapping useStore() (e.g. useAppStore())
```
👍

acusti reviewed (commented):
this looks good to me! there’s a handful of pnpm lockfile changes (primarily a new dependency on `astro@5.16.4` and a new optional dependency on `expo-router@6.0.10`), so hopefully those are expected, but definitely those docs changes look clear and correct.

acusti commented:
@IGassmann i submitted my approving review as a comment, but happy to submit an `Approve` review if that would be better, i just didn’t want to overstep

IGassmann committed:
docs: replace ambiguous state$ placeholder with concrete query example

The React component snippet in the reactivity system docs used a
`declare const state$: LiveQueryDef<number>` placeholder that didn't
clarify where the query definition came from. This replaces it with
a complete inline example showing queryDb() usage with a todos table,
making the documentation self-contained and easier to follow.

IGassmann committed:
docs: replace ambiguous state$ placeholder in Solid snippet

Similar to the React component update, this replaces the
`declare const state$: LiveQueryDef<number>` placeholder with
a concrete inline query example using queryDb() with a todos table,
making the documentation self-contained and easier to follow.

IGassmann committed:
refactor(docs): move solid snippets to framework-integrations/solid

Moves solid-integration/ to framework-integrations/solid/ for
consistency with react/, vue/, and svelte/ directory structure.
Updates all import paths accordingly.
